flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -1314,33 +1314,69 @@ static void genBodyOfTargetOp(
     for (mlir::Value val : valuesDefinedAbove) {
       mlir::Operation *valOp = val.getDefiningOp();
       assert(valOp != nullptr);
-      if (mlir::isMemoryEffectFree(valOp)) {
+
+      // NOTE: We skip BoxDimsOp's as the lesser of two evils is to map the
+      // indices separately, as the alternative is to eventually map the Box,
+      // which comes with a fairly large overhead comparatively. We could be
+      // more robust about this and check using a BackWardsSlice to see if we
+      // run the risk of mapping a box.
```

Comment:
Would this warrant an assert or something to make this explicitly visible should a code cause this to happen?

Reply:
I am a bit hesitant to add an assert, as it isn't an issue of correctness (at least it shouldn't be now after this patch and some testing), as it will work, it's just a bit more of an egregious copy now than it was before as the mappings gotten a little more complicated, so avoiding where possible is nice. Basically don't want to assert for perfectly valid and working code! It would be nice to be able to add performance warnings for these kinds of cases as opposed to flat ICEs/asserts, but I don't think flang has infrastructure for this sadly

Reply:
Ok. I'm fine with that.